pp_rework/ExamplePP.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on July 10, 2013
+
+@author: alfoa
+"""
+from __future__ import division, print_function , unicode_literals, absolute_import
+import warnings
+warnings.simplefilter('default', DeprecationWarning)
+
+#External Modules---------------------------------------------------------------
+import copy
+#External Modules End-----------------------------------------------------------
+
+#Internal Modules---------------------------------------------------------------
+from BaseClasses import BaseType
+from utils import InputData
+from Assembler import Assembler
+import MessageHandler
+#Internal Modules End-----------------------------------------------------------
+
+class ExamplePostProcessor(PostProcessorBase):
+  """
+    This is the base class for ExamplePostProcessor
+  """
+  @classmethod
+  def getInputSpecification(cls): # --> An interface should created for this method
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    inputSpecification = super(ExamplePostProcessor, cls).getInputSpecification()
+    return inputSpecification
```

Comment:
we may do not need this if we do not want to check the users post processor inputs

Reply:
Just to clarify, when we say "users" we are talking about future postprocessor developers, right?

Reply:
I mean future post processor plugin developers

Reply:
All postprocessors will follow this same API, though, whether "plugin" or standard, right?

Reply:
It seems to me it may too much for plugin developer. If we include this, the plugin developer should understand the InputData structure.